ci: Use remote pull/merge ref instead of local git merge

maflcko opened the pull request:
The merge strategy on the remote may be different than the local one. This may cause local merges to be different or fail completely. Fix this by using the result of the remote merge.

Fixes https://github.com/bitcoin/bitcoin/issues/26163

hebasto commented on .cirrus.yml:
```diff
@@ -30,8 +30,8 @@ base_template: &BASE_TEMPLATE
-    - git fetch $CIRRUS_REPO_CLONE_URL $CIRRUS_BASE_BRANCH
-    - git merge FETCH_HEAD  # Merge base to detect silent merge conflicts
+    - git fetch $CIRRUS_REPO_CLONE_URL "pull/${CIRRUS_PR}/merge"
+    - git checkout FETCH_HEAD  # Use merged changes to detect silent merge conflicts
```
Is comment correct? Cam `git checkout` really "_detect_ silent merge conflicts"?

MarcoFalke replied:
Yes, the silent conflicts will be detected by the following ci script (`make check` etc). This change isn't changing that

pameladuff replied:
Marcofaulke(polo) do you remember me? Please email me need to talk to you again thanks Pamm

Commit:
ci: Use remote pull/merge ref instead of local git merge

The merge strategy on the remote may be different than the local one.
This may cause local merges to be different or fail completely. Fix this
by using the result of the remote merge.

hebasto reviewed (approved):
ACK fad7281d7842f337932cf44e703fdd631230ddd6, I regularly use the same commands locally.

Interesting, that `git fetch origin pull/ID/merge` command was mentioned in GitHub's docs in the past. But I failed to find a reference to it now...

luke-jr reviewed (approved):
ACK

Checked that CI is getting the right thing checked out.
